hotham-simulator/src/action_state.rs

```diff
@@ -0,0 +1,34 @@
+use std::collections::HashMap;
+
+use openxr_sys::{Action, Path, FALSE};
+
+#[derive(Debug, Clone, Default)]
+/// Stores Action state, allowing the simulator to simulate input for an application.
+// A bit yuck to use u64 instead of Action, but it doesn't support Hash.. but whatever.
+pub struct ActionState {
+    boolean_actions: HashMap<u64, bool>,
+    bindings: HashMap<Path, u64>,
+}
```

Comment:
I think we should be able to [use the newtype pattern](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#using-the-newtype-pattern-to-implement-external-traits-on-external-types) to implement `Hash` for `Action`.

Reply:
We could, and that was my first thought, but I think that's unnecessary. We've got type safety anyway in the public API of `ActionState`, I don't think there's much to be gained otherwise.